src/registry/update_markdown.rs

```diff
@@ -25,13 +26,53 @@ pub struct RegistryUpdateMarkdownArgs {
     /// registry is located
     #[arg(short = 'd', long, default_value = "model")]
     pub registry_git_sub_dir: Option<String>,
+
+    /// Whether or not to run updates in dry-run mode.
+    #[arg(long, default_value = "false")]
+    pub dry_run: bool,
 }
 
 /// Update markdown files.
 pub(crate) fn command(
-    _log: impl Logger + Sync + Clone,
-    _cache: &Cache,
+    log: impl Logger + Sync + Clone,
+    cache: &Cache,
     args: &RegistryUpdateMarkdownArgs,
 ) {
-    println!("Args: {:#?}", args);
+    let registry = ResolvedSemconvRegistry::try_from_url(
+        args.registry.clone(),
+        args.registry_git_sub_dir.clone(),
+        cache,
+        log.clone(),
+    )
+    .unwrap_or_else(|e| {
+        panic!("Failed to resolve the semantic convention registry.\n{e}");
+    });
+    log.success("Registry resolved succesfully");
+    fn is_markdown(entry: &walkdir::DirEntry) -> bool {
+        let path = entry.path();
+        let extension = path.extension().unwrap_or_else(|| std::ffi::OsStr::new(""));
+        path.is_file() && extension == "md"
+    }
+    let operation = if args.dry_run {
+        "Validating"
+    } else {
+        "Updating"
+    };
+    let mut has_error = false;
+    for entry in walkdir::WalkDir::new(args.markdown_dir.clone())
+        .into_iter()
+        .filter_map(|e| match e {
+            Ok(v) if is_markdown(&v) => Some(v),
+            _ => None,
+        })
+    {
+        log.info(&format!("{}: ${}", operation, entry.path().display()));
+        if let Err(error) =
+            update_markdown(&entry.path().display().to_string(), &registry, args.dry_run)
+        {
+            has_error = true;
+            log.error(&format!("{error}"));
+        }
+    }
+    panic!("weaver registry update-markdown failed.")
```

Comment:
I don't understand this panic! at the end of the command.

Reply:
Fixed

Reply:
Yes either panic! or process::exit(1). (see https://doc.rust-lang.org/std/process/fn.exit.html)